lib/feed/view/feed_page.dart

```diff
@@ -415,6 +422,41 @@ class _FeedViewState extends State<FeedView> {
       ),
     );
   }
+
+  FutureOr<bool> _handleBack(bool stopDefaultButtonEvent, RouteInfo info) async {
+    FeedBloc feedBloc = context.read<FeedBloc>();
+
+    // See if we're at the top level of navigation
+    final canPop = Navigator.of(context).canPop();
+
+    // Get the desired post listing so we can check against current
+    final prefs = (await UserPreferences.instance).sharedPreferences;
+    final desiredPostListingType = PostListingType.values.byName(prefs.getString("setting_general_default_listing_type") ?? DEFAULT_LISTING_TYPE.name);
+    final currentPostListingType = feedBloc.state.postListingType;
+
+    // See if we're in a community
+    final currentCommunityId = feedBloc.state.communityId;
+
+    // If
+    // - We're at the top level of navigation AND
+    // - We're not on the desired listing type OR
+    // - We're on a community
+    // THEN navigate to the desired listing type
+    if (!canPop && (desiredPostListingType != currentPostListingType || currentCommunityId != null)) {
+      feedBloc.add(
+        FeedFetchedEvent(
+          sortType: feedBloc.state.sortType,
```

Comment:
Hmm, that's a great question! I was leaning towards leaving it as-is. Then I noticed that navigating to different feeds/communities via the sidebar resets the sort type to the default. I can't remember if it was always like that. Regardless, since I think the back button action should be roughly equivalent to navigating via the sidebar, I agree to reset the sort type. Done!